src/analysis/sum_integral_comparisons.lean

```diff
@@ -0,0 +1,165 @@
+an infinite sum to the improper integral of an antitone function. This file will eventually enable
+that.
+
+At the moment it contains two lemmas in this direction: `antitone_on.integral_le_sum` and
+`antitone_on.sum_le_integral` which can be paired with a `filter.tendsto` to estimate some errors.
+
+`TODO`: Add more lemmas to the API to directly address limiting issues
+
+## Main Results
+
+* `antitone_on.integral_le_sum`: The integral of an antitone function is at most the sum of its
+  values at integer steps aligning with the left-hand side of the interval
+* `antitone_on.sum_le_integral`: The sum of an antitone function along integer steps aligning with
+  the right-hand side of the interval is at most the integral of the function along that interval
+
+## Tags
+
+analysis, comparison, asymptotics
+-/
+
+open set measure_theory.measure_space
+open_locale big_operators
+
+lemma antitone_on.integral_le_sum {x₀ : ℝ} {a : ℕ} {f : ℝ → ℝ}
+  (hf : antitone_on f (Icc x₀ (x₀ + a))) :
+  ∫ x in x₀..(x₀ + a), f x ≤ ∑ i in finset.range a, f (x₀ + i) :=
+begin
+  have hint : ∀ (k : ℕ), k < a → interval_integrable f volume (x₀+k) (x₀ + (k + 1 : ℕ)),
+  { assume k hk,
+    refine (hf.mono _).interval_integrable,
+    rw interval_of_le,
+    { apply Icc_subset_Icc,
+      { simp only [le_add_iff_nonneg_right, nat.cast_nonneg] },
+      { simp only [add_le_add_iff_left, nat.cast_le, nat.succ_le_of_lt hk] } },
+    { simp only [add_le_add_iff_left, nat.cast_le, nat.le_succ] } },
+  calc
+  ∫ x in x₀..(x₀ + a), f x = ∑ i in finset.range a, ∫ x in (x₀+i)..(x₀+(i+1 : ℕ)), f x :
+    begin
+      convert (interval_integral.sum_integral_adjacent_intervals hint).symm,
+      simp only [nat.cast_zero, add_zero],
+    end
+  ... ≤ ∑ i in finset.range a, ∫ x in (x₀+i)..(x₀+(i+1)), f (x₀ + i) :
+    begin
+      apply finset.sum_le_sum (λ i hi, _),
+      have ia : i < a := finset.mem_range.1 hi,
+      refine interval_integral.integral_mono_on (by simp) (hint _ ia) (by simp) (λ x hx, _),
+      apply hf _ _ hx.1,
+      { simp only [ia.le, mem_Icc, le_add_iff_nonneg_right, nat.cast_nonneg, add_le_add_iff_left,
+          nat.cast_le, and_self] },
+      { refine mem_Icc.2 ⟨le_trans (by simp) hx.1, le_trans hx.2 _⟩,
+        simp only [add_le_add_iff_left, nat.cast_le, nat.succ_le_of_lt ia] },
+    end
+  ... = ∑ i in finset.range a, f (x₀ + i) : by simp
+end
+
+lemma antitone_on.integral_le_sum_Ico {a b : ℕ} {f : ℝ → ℝ} (hab : a ≤ b)
+  (hf : antitone_on f (set.Icc a b)) : ∫ x in a..b, f x ≤ ∑ x in finset.Ico a b, f x :=
+begin
+  have bb : b = (b - a) + a, { zify, ring, },
+  conv { to_rhs, rw [←zero_add a, bb], },
+  have bb' : (b : ℝ) = (a : ℝ) + ↑(b - a), { simp [hab], },
+  conv { to_lhs, rw bb', },
+  rw [←finset.sum_Ico_add, nat.Ico_zero_eq_range],
+  conv
+  { to_rhs, congr, skip, funext, rw nat.cast_add a x, },
+  apply antitone_on.integral_le_sum,
+  rwa ←bb',
+end
+
+lemma antitone_on.sum_le_integral
```

Comment:
Isn't it possible to deduce this lemma from the other one by applying it to `-f (-x)` instead of redoing the proof? By the way, would it also be possible to obtain the analogous results for monotone functions, again by applying the previous ones for `-f x` or `f (-x)`?

Reply:
Potentially! Do you know if lemmas like these:

```lean
lemma antitone.monotone_of_neg_arg { f : ℝ → ℝ } (hf : antitone f) :
  monotone (λ x, f (-x))
```

already exist in the library? I couldn't find them, but could certainly add them.

Reply:
Also, I have now spent a bit of time on this and I've concluded that while your suggestion is the "correct" argument, it creates an enormous headache for two reasons:
* reversing sums is extremely hard
* manipulating natural numbers is hard

Fundamentally, this comes down to the fact that natural number subtraction is _extremely_ difficult to work with in mathlib. As an example of where I got you can see already a lot of subtraction sadness is being worked around:

https://gist.github.com/khwilson/3e766bed21bd65b920c4323b0157b0db

Reply:
Yes, I agree that this will be more painful than copy/paste/fix